core/shared/src/main/scala-3/zio/ZLayerCompanionVersionSpecific.scala

```diff
@@ -38,4 +40,24 @@ trait ZLayerCompanionVersionSpecific {
    */
   def makeSome[R0, R] =
     new WireSomePartiallyApplied[R0, R]
+
+  /**
+   * Derives a simple layer for a case class given as a type parameter.
+   * {{{
+   * case class Car(engine: Engine, wheels: Wheels)
+   * val derivedLayer: ZLayer[Engine & Wheels, Nothing, Car] = ZLayer.deriveLayer[Car]
+   * // equivalent to:
+   * val manualLayer: ZLayer[Engine with Wheels, Nothing, Car] = ZLayer {
+   *   for {
+   *     engine <- ZIO.service[Engine]
+   *     wheels <- ZIO.service[Wheels]
+   *   } yield Car(engine, wheels)
+   * }
+   * }}}
+   *
+   */
+   inline def deriveLayer[A](
```

Comment:
I think this could just be `derive` since it is already an operator on the `ZLayer` companion object.